addons/l10n_in_tcs_tds/__init__.py

```diff
@@ -0,0 +1,47 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from . import models
+
+from odoo import api, SUPERUSER_ID
+
+import logging
+
+_logger = logging.getLogger(__name__)
+
+
+def load_taxes(env):
+    in_chart_template = env.ref('l10n_in.indian_chart_template_standard')
+    for company in env['res.company'].search([('chart_template_id', '=', in_chart_template.id)]):
+        try:
+            with env.cr.savepoint():
+                tax_template_ids = env['ir.model.data'].search([
+                    ('module', '=', 'l10n_in_tcs_tds'),
+                    ('model', '=', 'account.tax.template'),
+                    ]).mapped('res_id')
+                generated_tax_res = env['account.tax.template'].browse(tax_template_ids)._generate_tax(company)
+                taxes_ref = generated_tax_res['tax_template_to_tax']
+        except Exception:
```

Comment:
what can go wrong?  I don't think the savepoint and try..except is needed

Reply:
The max chance is that the same tax is available in DB so we need try..except

Reply:
If they already have it, maybe they should not install this module...  I suppose the xml_ids won't be double, also because the module name is in there too and you can create the same taxes twice. Only the name should be unique.  It is also not good to catch an Exception that can be anything else than the uniqueness constraint.

Reply:
For the reasons explained by @jco-odoo , I'd rather avoid the try...except by searching if there exists an XML_id for the tax belonging to the company. Such an xml_id will be created the first time by  _create_records_with_xmlid(). :point_up:

Reply:
when system try _create_records_with_xmlid but same time account code is used by the customer and we have a constraint in the account and tax to not repeat the same.

if there is an error in the hook then I think the module install failed.